### PR TITLE
Refactoring Siddhi GPL GEO extension

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

**Review by AnuGayan — COMMENTED** [review]
Please review and merge

File: component/pom.xml [hunk 1]
```diff
@@ -24,14 +24,13 @@
     <parent>
         <groupId>org.wso2.extension.siddhi.gpl.execution.geo</groupId>
         <artifactId>siddhi-gpl-execution-geo-parent</artifactId>
-        <version>3.1.3-SNAPSHOT</version>
+        <version>4.0.0</version>
```
Comment (suhothayan) [inline component/pom.xml]: it should be 4.0.0-SNAPSHOT

File: component/pom.xml [hunk 2]
```diff
@@ -24,14 +24,13 @@
...
         <relativePath>../pom.xml</relativePath>
     </parent>
 
     <artifactId>siddhi-gpl-execution-geo</artifactId>
     <packaging>bundle</packaging>
-    <name>WSO2 Siddhi Execution Extension - Geo</name>
-    <url>http://wso2.org</url>
+    <name>Siddhi Execution Extension - Geo</name>
```
Comment (suhothayan) [inline component/pom.xml]: Siddhi "GPL" Execution Extension - Geo  missed GPL

File: component/pom.xml [hunk 3]
```diff
@@ -92,18 +91,19 @@
...
             <artifactId>gson</artifactId>
         </dependency>
         <dependency>
             <groupId>org.wso2.orbit.org.geotools</groupId>
             <artifactId>gt-geojson</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>6.8</version>
```
Comment (suhothayan) [inline component/pom.xml]: Move version to properties

File: feature/pom.xml [hunk 4]
```diff
@@ -21,15 +21,14 @@
...
         <relativePath>../pom.xml</relativePath>
     </parent>
 
     <artifactId>siddhi-gpl-execution-geo-feature</artifactId>
     <packaging>pom</packaging>
-    <name>WSO2 Siddhi Execution Extension - Geo Feature</name>
-    <url>http://wso2.org</url>
-    <description>This feature contains Siddhi extension feature for Geo.</description>
+    <name>Siddhi Execution Extension - Geo Feature</name>
+    <description>This feature contains Siddhi extension feature for Geo.kk</description>
```
Comment (suhothayan) [inline feature/pom.xml]: feature should be deleted

File: pom.xml [hunk 5]
```diff
@@ -47,11 +47,11 @@
     </profiles>
 
     <properties>
-        <siddhi.version>3.1.2</siddhi.version>
+        <siddhi.version>4.0.0-M7</siddhi.version>
         <log4j.version>1.2.17.wso2v1</log4j.version>
         <gson.version>2.2.4</gson.version>
         <geotools.version>13.6.0.wso2v1</geotools.version>
-        <junit.version>4.10</junit.version>
+        <junit.version>4.12</junit.version>
```
Comment (suhothayan) [inline pom.xml]: remove junit, use testng

File: pom.xml [hunk 6]
```diff
@@ -113,9 +113,9 @@
                 </exclusions>
             </dependency>
             <dependency>
-                <groupId>junit</groupId>
-                <artifactId>junit</artifactId>
-                <version>${junit.version}</version>
+                <groupId>org.testng</groupId>
+                <artifactId>testng</artifactId>
+                <version>6.8</version>
```
Comment (suhothayan) [inline pom.xml]: move version to properties

File: component/pom.xml (hunk 1 above)
Comment (suhothayan) [reply component/pom.xml]: Component artifactId should not have "-parent" part in the name componet.

**Review by suhothayan — CHANGES_REQUESTED** [review]

File: component/pom.xml (hunk 1 above)
Comment (AnuGayan) [reply component/pom.xml]: @suhothayan Fixed the version.  
"-parent" in the artifactId is just to refer to the parent pom.

File: component/pom.xml (hunk 2 above)
Comment (AnuGayan) [reply component/pom.xml]: @suhothayan Fixed

File: component/pom.xml (hunk 3 above)
Comment (AnuGayan) [reply component/pom.xml]: @suhothayan Fixed

File: feature/pom.xml (hunk 4 above)
Comment (AnuGayan) [reply feature/pom.xml]: @suhothayan Fixed

File: pom.xml (hunk 5 above)
Comment (AnuGayan) [reply pom.xml]: @suhothayan Fixed

File: pom.xml (hunk 6 above)
Comment (AnuGayan) [reply pom.xml]: @suhothayan Fixed

File: component/pom.xml [hunk 7]
```diff
@@ -19,19 +19,19 @@
...
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <modelVersion>4.0.0</modelVersion>
+
 
     <parent>
         <groupId>org.wso2.extension.siddhi.gpl.execution.geo</groupId>
         <artifactId>siddhi-gpl-execution-geo-parent</artifactId>
-        <version>3.1.3-SNAPSHOT</version>
+        <version>4.0.0-SNAPSHOT</version>
```
Comment (tishan89) [inline component/pom.xml]: make it 1.0.0 snapshot

File: feature/pom.xml [hunk 8]
```diff
@@ -16,19 +16,18 @@
 -->
```
Comment (tishan89) [inline feature/pom.xml]: remove feature all together

File: pom.xml [hunk 9]
```diff
@@ -18,20 +18,20 @@
...
-        <version>1</version>
+        <groupId>org.wso2</groupId>
+        <version>5</version>
     </parent>
+    <modelVersion>4.0.0</modelVersion>
 
     <groupId>org.wso2.extension.siddhi.gpl.execution.geo</groupId>
     <artifactId>siddhi-gpl-execution-geo-parent</artifactId>
-    <version>3.1.3-SNAPSHOT</version>
+    <version>4.0.0-SNAPSHOT</version>
```
Comment (tishan89) [inline pom.xml]: 1.0.0-SNAPSHOT

**Review by tishan89 — CHANGES_REQUESTED** [review]

File: feature/pom.xml (hunk 8 above)
Comment (AnuGayan) [reply feature/pom.xml]: @tishan89 Done

**Review by tishan89 — APPROVED** [review]

**Pull request merged**